src/App.tsx

```diff
@@ -1,15 +1,47 @@
-import React from 'react';
+import React, { useState } from 'react';
+import debounce from 'lodash.debounce';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [search, setSearch] = useState('');
+  const [filteredPeople, setFilteredPeople] = useState(peopleFromServer);
+  const [isActive, setIsActive] = useState(false);
+  const [item, setItem] = useState<Person | null>(null);
+
+  const handleFocus = () => {
+    setIsActive(!isActive);
+  };
+
+  const handleSearchChange = debounce((query: string) => {
+    setFilteredPeople(
+      peopleFromServer.filter(person =>
+        person.name.toLowerCase().includes(query.toLowerCase()),
+      ),
+    );
+  }, 300);
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const query = event.target.value;
+
+    setSearch(query);
+    handleSearchChange(query);
+  };
+
+  const handleItemClick = (people: Person) => {
+    setItem(people);
+    setSearch(people.name);
+    setIsActive(false);
+  };
 
   return (
     <div className="container">
       <main className="section is-flex is-flex-direction-column">
         <h1 className="title" data-cy="title">
-          {`${name} (${born} - ${died})`}
+          {!item
+            ? 'No selected person'
+            : `${item?.name} (${item?.born} - ${item?.died})`}
```

Comment:
Consider will be better if you move this checking to the helper variable and use it here